### PR TITLE
Add simple custom outputs (functions of other outputs) to all solvers

### PR DESCRIPTION
**Pull request by pcarruscag**
## Proposed Changes
Adds the simplest type of custom output to all solvers. Add coverage for custom functions in some discrete adjoint cases.

## PR Checklist
- [X] I am submitting my contribution to the develop branch.
- [X] My contribution generates no new compiler warnings (try with --warnlevel=3 when using meson).
- [X] My contribution is commented and consistent with SU2 style (https://su2code.github.io/docs_v7/Style-Guide/).
- [X] I have added a test case that demonstrates my contribution, if necessary.
- [X] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp), if necessary.

### CODE REVIEW

File: TestCases/species_transport/venturi_primitive_3species/DAspecies3_primitiveVenturi.cfg [hunk 1]
```diff
@@ -14,6 +14,11 @@
 SOLVER= INC_RANS
 KIND_TURB_MODEL= SST
 %
+CUSTOM_OUTPUTS= 'avg_species_0 : AreaAvg{SPECIES[0]}[outlet];\
+                 avg_species_1 : AreaAvg{SPECIES[1]}[outlet];\
+                 var_species : AreaAvg{pow(SPECIES[0] - avg_species_0, 2) +\
+                                       pow(SPECIES[1] - avg_species_1, 2)}[outlet]'
+CUSTOM_OBJFUNC= 'var_species'
 OBJECTIVE_FUNCTION= SURFACE_SPECIES_VARIANCE
```
Comment (pcarruscag) [inline TestCases/species_transport/venturi_primitive_3species/DAspecies3_primitiveVenturi.cfg:22]:
```suggestion
OBJECTIVE_FUNCTION= CUSTOM_OBJFUNC
```

File: SU2_CFD/src/output/COutput.cpp [hunk 2]
```diff
@@ -2287,6 +2287,46 @@ void COutput::SetCustomOutputs(const CConfig* config) {
...
+        output.varIndices.push_back(output.varIndices.size());
+        output.otherOutputs.push_back(GetPtrToHistoryOutput(var));
+        if (output.otherOutputs.back() == nullptr) {
+          if (!adjoint) {
+            // In primal mode all functions must be valid.
+            SU2_MPI::Error("Invalid history output (" + var + ") used in function " + output.name, CURRENT_FUNCTION);
+          } else {
+            if (rank == MASTER_NODE) {
+              std::cout << "Info: Ignoring function " + output.name + " because it may be used by the primal/adjoint "
+                          "solver.\n      If the function is ignored twice it is invalid." << std::endl;
```
Comment (pcarruscag) [inline SU2_CFD/src/output/COutput.cpp:2313]:
```suggestion
                           "solver.\n      If the function is ignored twice it is invalid." << std::endl;
```

File: SU2_CFD/src/output/CFlowOutput.cpp [hunk 3]
```diff
@@ -796,23 +796,38 @@ void CFlowOutput::ConvertVariableSymbolsToIndices(const CPrimitiveIndices<unsign
...
     if (output.otherOutputs.back() == nullptr) {
-      SU2_MPI::Error("Invalid history output or solver variable (" + var + ") used in function " + output.name +
-                     "\nValid solvers variables:\n" + knownVariables.str(), CURRENT_FUNCTION);
+      if (!allowSkip) {
+        SU2_MPI::Error("Invalid history output or solver variable (" + var + ") used in function " + output.name +
+                      "\nValid solvers variables:\n" + knownVariables.str(), CURRENT_FUNCTION);
+      } else {
+        if (rank == MASTER_NODE) {
+          std::cout << "Info: Ignoring function " + output.name + " because it may be used by the primal/adjoint "
+                        "solver.\n      If the function is ignored twice it is invalid." << std::endl;
```
Comment (pcarruscag) [inline SU2_CFD/src/output/CFlowOutput.cpp:805]:
```suggestion
                       "solver.\n      If the function is ignored twice it is invalid." << std::endl;
```

File: SU2_CFD/src/output/CFlowOutput.cpp [hunk 4]
```diff
@@ -796,23 +796,38 @@ void CFlowOutput::ConvertVariableSymbolsToIndices(const CPrimitiveIndices<unsign
     output.varIndices.back() += output.otherOutputs.size();
     output.otherOutputs.push_back(GetPtrToHistoryOutput(var));
     if (output.otherOutputs.back() == nullptr) {
-      SU2_MPI::Error("Invalid history output or solver variable (" + var + ") used in function " + output.name +
-                     "\nValid solvers variables:\n" + knownVariables.str(), CURRENT_FUNCTION);
+      if (!allowSkip) {
+        SU2_MPI::Error("Invalid history output or solver variable (" + var + ") used in function " + output.name +
+                      "\nValid solvers variables:\n" + knownVariables.str(), CURRENT_FUNCTION);
```
Comment (pcarruscag) [inline SU2_CFD/src/output/CFlowOutput.cpp:801]:
```suggestion
                       "\nValid solvers variables:\n" + knownVariables.str(), CURRENT_FUNCTION);
```

File: TestCases/species_transport/venturi_primitive_3species/DAspecies3_primitiveVenturi.cfg [hunk 5]
```diff
@@ -14,7 +14,12 @@
 SOLVER= INC_RANS
 KIND_TURB_MODEL= SST
 %
-OBJECTIVE_FUNCTION= SURFACE_SPECIES_VARIANCE
+CUSTOM_OUTPUTS= 'avg_species_0 : AreaAvg{SPECIES[0]}[outlet];\
+                 avg_species_1 : AreaAvg{SPECIES[1]}[outlet];\
+                 var_species : AreaAvg{pow(SPECIES[0] - avg_species_0, 2) +\
+                                       pow(SPECIES[1] - avg_species_1, 2)}[outlet]'
+CUSTOM_OBJFUNC= 'var_species'
+OBJECTIVE_FUNCTION= CUSTOM_OBJFUNC
```
Comment (bigfooted) [inline TestCases/species_transport/venturi_primitive_3species/DAspecies3_primitiveVenturi.cfg:22]: Thanks, another step that makes life easier for all of us :-)

**Pull request merged**